packages/quantic/force-app/main/default/lwc/quanticStandaloneSearchBox/__tests__/quanticStandaloneSearchBox.test.js

```diff
@@ -0,0 +1,188 @@
+  () => ({
+    default: jest.fn(),
+  }),
+  {virtual: true}
+);
+
+mockHeadlessLoader.loadDependencies = () =>
+  new Promise((resolve) => {
+    resolve();
+  });
+
+let isInitialized = false;
+
+const exampleEngine = {
+  id: 'engineId',
+};
+
+const functionsMocks = {
+  buildStandaloneSearchBox: jest.fn(() => ({
+    state: {},
+    subscribe: functionsMocks.subscribe,
+  })),
+  subscribe: jest.fn((cb) => {
+    cb();
+    return functionsMocks.unsubscribe;
+  }),
+  unsubscribe: jest.fn(() => {}),
+};
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  placeholder: null,
+  withoutSubmitButton: false,
+  numberOfSuggestions: 7,
+  textarea: false,
+  disableRecentQueries: false,
+  keepFiltersOnSearch: false,
+  redirectUrl: '/global-search/%40uri',
+};
+
+function createTestComponent(options = defaultOptions) {
+  prepareHeadlessState();
+  const element = createElement('c-quantic-standalone-search-box', {
+    is: QuanticStandaloneSearchBox,
+  });
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+  document.body.appendChild(element);
+  return element;
+}
+
+function prepareHeadlessState() {
+  // @ts-ignore
+  global.CoveoHeadless = {
+    buildStandaloneSearchBox: functionsMocks.buildStandaloneSearchBox,
+  };
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  // eslint-disable-next-line @lwc/lwc/no-async-operation
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function mockSuccessfulHeadlessInitialization() {
+  // @ts-ignore
+  mockHeadlessLoader.initializeWithHeadless = (element, _, initialize) => {
+    if (element instanceof QuanticStandaloneSearchBox && !isInitialized) {
+      isInitialized = true;
+      initialize(exampleEngine);
+    }
+  };
+}
+
+function cleanup() {
+  // The jsdom instance is shared across test cases in a single file so reset the DOM
+  while (document.body.firstChild) {
+    document.body.removeChild(document.body.firstChild);
+  }
+  jest.clearAllMocks();
+  isInitialized = false;
+}
+
+describe('c-quantic-standalone-search-box', () => {
+  beforeEach(() => {
+    getHeadlessConfiguration.mockResolvedValue(defaultHeadlessConfiguration);
+    mockSuccessfulHeadlessInitialization();
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  it('construct itself without throwing', () => {
+    expect(() => createTestComponent()).not.toThrow();
+  });
+
+  describe('controller initialization', () => {
+    it('should subscribe to the headless state changes', async () => {
+      createTestComponent();
+      await flushPromises();
+
+      expect(functionsMocks.subscribe).toHaveBeenCalledTimes(1);
+    });
+
+    describe('when the current page reference changes', () => {
+      beforeAll(() => {
+        Object.defineProperty(window, 'location', {
+          writable: true,
+          value: {href: nonStandaloneURL},
+        });
```

Comment:
Interesting, can you explain this? perhaps a comment and a reference where you found the doc for this?

Reply:
Also why do you need this?

Reply:
this is because in order to test that the `keep-filters-on-search` is properly passed in the `c-quantic-search-box`, you need `isStandalone` to be false. 

The `isStandalone` property is set like this whenever the page ref changes:
`this.isStandalone = !window.location.href.includes(this.redirectUrl);`

When the the URL does not include the redirection URL. So this weird logic is to set a different URL in the window location and force the situation where the we render the `c-quantic-search-box`

So you need to set this property in order to be able to emit the wire event for the test:
`CurrentPageReference.emit({url: nonStandaloneURL});`

Reply:
For your first comment, this is the ref I used: https://stackoverflow.com/questions/54021037/how-to-mock-window-location-href-with-jest-vuejs

Its to mock the window location in jest.